sky/serve/service_spec.py

```diff
@@ -220,8 +226,14 @@ def add_if_not_none(section, key, value, no_empty: bool = False):
 
     def probe_str(self):
         if self.post_data is None:
-            return f'GET {self.readiness_path}'
-        return f'POST {self.readiness_path} {json.dumps(self.post_data)}'
+            method = f'GET {self.readiness_path}'
+        else:
+            method = f'POST {self.readiness_path} {json.dumps(self.post_data)}'
+        if self.readiness_headers is None:
+            headers = ''
+        else:
+            headers = f' with headers {json.dumps(self.readiness_headers)}'
```

Comment:
nit: Showing the authorization token directly in the log may not be very safe. We may want to redact the value of the headers?

Reply:
Good point! Changed to ` with Authotization` now.